### ISSUE TITLE
oversaturated dark areas playing HDR 10 bit

### ISSUE THREAD

**Issue by sfDuhNinja** [iina/iina#2233]
**System and IINA version:**

- macOS 10.14.2 (18C54)
- IINA 1.0.1 Build 103

**Expected behavior:**
The dark areas should have details. 
**Actual behavior:**
The dark areas does not retain any details. In the latest vlc the picture is pleasant to look at. Also, the reds seems to be oversaturated as well. All this while watching it on a MBP 13 2018.

_please look at the beard and hair_

IINA
<img width="1440" alt="iina" src="https://user-images.githubusercontent.com/27433856/51384471-1f0a0100-1b25-11e9-9b52-2c2f411cfe9c.png">

VLC
<img width="1440" alt="vlc" src="https://user-images.githubusercontent.com/27433856/51384578-727c4f00-1b25-11e9-8e9e-0dbde1b7fd7c.png">

MediaInfo

Video
ID                                       : 1
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Commercial name                          : HDR10
Format profile                           : Main 10@L5.1@High
Codec ID                                 : V_MPEGH/ISO/HEVC
Duration                                 : 1 h 9 min
Width                                    : 3 840 pixels
Height                                   : 2 160 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0 (Type 2)
Bit depth                                : 10 bits
Writing library                          : x265 2.9+8-27d8424c799d:[Windows][MSVC 1900][64 bit] 10bit
Encoding settings                        : cpuid=1176575 / frame-threads=6 / numa-pools=36 / wpp / no-pmode / no-pme / no-psnr / no-ssim / log-level=2 / input-csp=1 / input-res=3840x2160 / interlace=0 / total-frames=100040 / level-idc=51 / high-tier=1 / uhd-bd=0 / ref=4 / no-allow-non-conformance / repeat-headers / annexb / aud / hrd / info / hash=0 / no-temporal-layers / no-open-gop / min-keyint=24 / keyint=240 / gop-lookahead=0 / bframes=4 / b-adapt=2 / b-pyramid / bframe-bias=0 / rc-lookahead=25 / lookahead-slices=4 / scenecut=40 / radl=0 / no-intra-refresh / ctu=64 / min-cu-size=8 / rect / no-amp / max-tu-size=32 / tu-inter-depth=1 / tu-intra-depth=1 / limit-tu=0 / rdoq-level=2 / dynamic-rd=0.00 / no-ssim-rd / signhide / no-tskip / nr-intra=0 / nr-inter=0 / no-constrained-intra / no-strong-intra-smoothing / max-merge=3 / limit-refs=3 / limit-modes / me=3 / subme=3 / merange=57 / temporal-mvp / weightp / no-weightb / no-analyze-src-pics / deblock=-3:-3 / no-sao / no-sao-non-deblock / rd=4 / no-early-skip / rskip / no-fast-intra / no-tskip-fast / no-cu-lossless / no-b-intra / no-splitrd-skip / rdpenalty=0 / psy-rd=2.00 / psy-rdoq=1.00 / no-rd-refine / no-lossless / cbqpoffs=0 / crqpoffs=0 / rc=crf / crf=17.0 / qcomp=0.60 / qpstep=4 / stats-write=0 / stats-read=0 / vbv-maxrate=160000 / vbv-bufsize=160000 / vbv-init=0.9 / crf-max=0.0 / crf-min=0.0 / ipratio=1.40 / pbratio=1.30 / aq-mode=1 / aq-strength=1.00 / cutree / zone-count=0 / no-strict-cbr / qg-size=32 / no-rc-grain / qpmax=69 / qpmin=0 / no-const-vbv / sar=1 / overscan=0 / videoformat=5 / range=0 / colorprim=9 / transfer=16 / colormatrix=9 / chromaloc=1 / chromaloc-top=2 / chromaloc-bottom=2 / display-window=0 / master-display=G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(40000000,50) / max-cll=735,91 / min-luma=0 / max-luma=1023 / log2-max-poc-lsb=8 / vui-timing-info / vui-hrd-info / slices=1 / no-opt-qp-pps / no-opt-ref-list-length-pps / no-multi-pass-opt-rps / scenecut-bias=0.05 / no-opt-cu-delta-qp / no-aq-motion / hdr / hdr-opt / no-dhdr10-opt / no-idr-recovery-sei / analysis-reuse-level=5 / scale-factor=0 / refine-intra=0 / refine-inter=0 / refine-mv=0 / no-limit-sao / ctu-info=0 / no-lowpass-dct / refine-mv-type=0 / copy-pic=1 / max-ausize-factor=1.0 / no-dynamic-refine / no-single-sei
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.2020
Transfer characteristics                 : PQ
Matrix coefficients                      : BT.2020 non-constant
Mastering display color primaries        : Display P3
Mastering display luminance              : min: 0.0050 cd/m2, max: 4000 cd/m2
Maximum Content Light Level              : 735 cd/m2
Maximum Frame-Average Light Level        : 91 cd/m2
Statistics Tags Issue                    : no_variable_data 1970-01-01 00:00:00 / no_variable_data 2010-02-22 21:41:29
FromStats_BitRate                        : 24343148
FromStats_Duration                       : 01:09:32.502000000
FromStats_FrameCount                     : 100040
FromStats_StreamSize                     : 12696479367
<br>

**How often does this happen?**
With any HDR 10 bit blu ray video I tried.

**Comment by LeoNatan** [comment]
I see this too, and it wasn't like this before. Seems like a regression in 1.0.1.

**Comment by LeoNatan** [comment]
0.0.15
<img width="1072" alt="screen shot 2019-01-20 at 05 04 37" src="https://user-images.githubusercontent.com/2270433/51434763-0ad31a80-1c71-11e9-8c70-d3655cc2dc96.png">

Latest
<img width="1072" alt="screen shot 2019-01-20 at 05 04 59" src="https://user-images.githubusercontent.com/2270433/51434766-132b5580-1c71-11e9-8376-7a6fdd5ec63c.png">

Clearly a regression.

**Comment by inflation** [comment]
Since previous versions didn't do color conversion correctly I don't know the "correct" color. Which color space are you using on the monitor?

**Comment by inflation** [comment]
@LeoNatan Do you have `icc-profile-auto` in mpv's config? That would enable the correct color conversion.

**Comment by LeoNatan** [comment]
I am using the default Apple color profile for the MBP.

Tone mapping is a separate issue from color space. Perhaps when you added support for that, you broke tone mapping?

**Comment by LeoNatan** [comment]
@inflation Where do I add that?

**Comment by inflation** [comment]
I see identical results on my SRGB monitor between mpv and IINA. 

You can set `icc-profile` in Advanced settings to a invalid file. That could potentially convert anything to rec709.

**Comment by LeoNatan** [comment]
Yes, but what broke tone mapping?

BTW, I am testing this on a 2018 MBP, which has a wider color gamut than sRGB. Still no reason to break tone mapping.

**Comment by LeoNatan** [comment]
Setting `icc-profile` to "zzadasdad" doesn't help.

**Comment by LeoNatan** [comment]
https://github.com/iina/iina/issues/1994

People have reported this in the past. Why was that issue closed?

**Comment by inflation** [comment]
Yeah, it's an mpv change between the two version. You can try set `hdr-compute-peak` to `no`.

**Comment by LeoNatan** [comment]
Ha, that seems to help! So mpv changed that default value to YES?

I will read through that "subpar tone mapping" issue tomorrow to see if there is anything interesting there.

Thanks for the quick help. Consider changing that to "NO" in IINA as default.

**Comment by inflation** [comment]
Another thing you need to tweek if you set `hdr-compute-peak` to `no` is the algorithm, i.e., `tone-mapping`.

`reinhard` preserves overall brightness in trade of details and color correctness:
![image](https://user-images.githubusercontent.com/2375962/51435104-ec076080-1c3d-11e9-84ec-ab57ba25d0bc.png)

`hable` (default) gives better details in dark and light area but will darken/desaturate everything (so works fine if you compute the brightness):
![image](https://user-images.githubusercontent.com/2375962/51435106-ee69ba80-1c3d-11e9-85a9-ba0bcc86030b.png)

**Comment by inflation** [comment]
Yeah, that what I mean by "trading off". So before mpv merges the PR, the best you can do is playing around with those parameters.

**Comment by LeoNatan** [comment]
I'm not sure the two are related. This banding seems like an issue of color management.

**Comment by adamwinn** [comment]
I'm seeing this too.   Movist Pro looks way better

**Comment by IshanArya** [comment]
Having same problem. Dark scenes have detail completely removed.

<img width="1440" alt="Screen Shot 2019-07-21 at 9 44 46 PM" src="https://user-images.githubusercontent.com/10377398/61600842-d11fa400-ac00-11e9-8c81-a66f8e4c074c.png">

**Comment by ajmalabu** [comment]
Hey guys! I recently downloaded an HDR sample test video (link: `http://ogrfi3uyt7.com/w24bv54m8v?lbcnmbj=57&refer=http%3A%2F%2Fuhdsample.com%2F43-download-sony-bravia-make-glass-oled-4k-uhd-hdr10-60-fps-demo-video.html&kw=%5B%22sony%22%2C%22bravia%22%2C%22make%22%2C%22glass%22%2C%22oled%22%2C%22promotional%22%2C%22video%22%2C%22in%22%2C%22ultra-hd%22%2C%22hdr%22%2C%22hdr10%22%5D&key=f278b01e507050e65f7f845049826219&scrWidth=1680&scrHeight=1050&tz=3&ship=&pst=&v=19.11.v.7&res=7.13&dev=r`)

I played it on both IINA and QuickTime, and the results are as follows:

<img width="966" alt="Screen Shot 2019-11-20 at 5 17 59 PM" src="https://user-images.githubusercontent.com/20157320/69246920-a186a580-0bba-11ea-8f61-9e52b500a5f6.png">

Just curious, why exactly is this happening?

**Comment by qinyuhang** [comment]
I have the same experience.
Here is a compare between Moviest Pro and IINA
Left is Moviest, the right image is from IINA
<img width="1643" alt="企业微信截图_a1b8370e-6471-490f-8986-1ea525dcc90b" src="https://user-images.githubusercontent.com/6239057/86939115-bfb60b80-c173-11ea-8e41-4cde4e25a1cd.png">

**Comment by MtPython** [comment]
Still same issue/question as @ajmalabu had, the mpv seems to be slightly brighter, but miles away from QuickTime. VLC has the darkest picture overall.

**Comment by 95tuanle** [comment]
I'm also having the same problem with latest version of IINA

**Comment by gregcotten** [comment]
@ajmalabu Your link to uhdsample.com takes me to a malware site.
